mesheryctl/internal/cli/root/system/restart.go

```diff
@@ -31,12 +38,56 @@ var restartCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) error {
 		log.Info("Restarting Meshery...")
 
-		if err := stop(); err != nil {
-			return errors.Wrap(err, utils.SystemError("Failed to restart Meshery"))
+		// Get viper instance used for context
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return errors.Wrap(err, "error processing config")
+		}
+		// get the platform, channel and the version of the current context
+		// if a temp context is set using the -c flag, use it as the current context
+		currCtx, err := mctlCfg.SetCurrentContext(tempContext)
+		if err != nil {
+			return err
 		}
+		currPlatform := currCtx.Platform
+
+		switch currPlatform {
+		case "docker":
+			if err := stop(); err != nil {
+				return errors.Wrap(err, utils.SystemError("Failed to restart Meshery"))
+			}
+
+			if err := start(); err != nil {
+				return errors.Wrap(err, utils.SystemError("Failed to restart Meshery"))
+			}
+
+		case "kubernetes":
+			// create an kubernetes client
+			client, err := meshkitkube.New([]byte(""))
+
+			if err != nil {
+				return err
+			}
+
+			// Create a pod interface for the MesheryNamespace
+			podInterface := client.KubeClient.CoreV1().Pods(utils.MesheryNamespace)
+
+			// List the pods in the MesheryNamespace
+			podList, err := podInterface.List(context.TODO(), v1.ListOptions{})
+			if err != nil {
+				return err
+			}
 
-		if err := start(); err != nil {
-			return errors.Wrap(err, utils.SystemError("Failed to restart Meshery"))
+			// List all the pods similar to kubectl get pods -n MesheryNamespace
+			for _, pod := range podList.Items {
+				// Get the values from the pod status
+				name := pod.GetName()
+				log.Debug("Restarting pod ", name)
```

Comment:
@navendu-pottekkat here.

Reply:
Only shown if `-v` flag is present, yeah?

Reply:
Yes. @leecalcote

Reply:
“Restarting” or “Deleting”?

Reply:
Actually, it is deleting but pods get restarted when they get deleted in k8s.